proxywasm/proxytest/root.go

```diff
@@ -166,8 +166,12 @@ func (r *rootHostEmulator) ProxyGetSharedData(keyData *byte, keySize int,
 // impl internal.ProxyWasmHost
 func (r *rootHostEmulator) ProxySetSharedData(keyData *byte, keySize int,
 	valueData *byte, valueSize int, cas uint32) internal.Status {
-	key := internal.RawBytePtrToString(keyData, keySize)
-	value := internal.RawBytePtrToByteSlice(valueData, valueSize)
+	// Copy data provided by plugin to keep ownership within host. Otherwise, when
```

Comment:
Phew, this was a long debug. I didn't check if there's anywhere else in the emulator directly referencing plugin memory, but this was the only one that wasn't passing the wasm tests. Anyways confirm this makes sense